.github/workflows/linux-qe.yml

```diff
@@ -0,0 +1,150 @@
+        uses: actions/download-artifact@v4
+        with:
+          name: crc-e2e-linux-arm64
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ github.token }}
+
+      - name: Install CRC on host
+        working-directory: ${{ env.timestamp }}
+        run: |
+          podman run --rm -d --privileged --name crc-linux-install \
+            -e TARGET_HOST=$(cat host) \
+            -e TARGET_HOST_USERNAME=$(cat username) \
+            -e TARGET_HOST_KEY_PATH=/data/id_rsa \
+            -e BASTION_HOST_USERNAME=$(cat bastion_username) \
+            -e BASTION_HOST=$(cat bastion_host) \
+            -e TARGET_FOLDER=crc-support \
+            -e TARGET_CLEANUP='false' \
+            -e OUTPUT_FOLDER=/data \
+            -e DEBUG='true' \
+            -e SSH_AUTH_SOCK=$(cat ssh_auth_sock) \
+            -v "$(cat ssh_auth_sock):$(cat ssh_auth_sock)" \
+            -v ${PWD}:/data:z \
+            -v ${PWD}/crc:/opt/crc-support/crc:z \
+            quay.io/rhqp/crc-support:v0.5.1-linux crc-support/install.sh crc-support/crc
+          podman logs -f crc-linux-install
+
+      - name: Run CRC e2e test
+        working-directory: ${{ env.timestamp }}
+        run: |
+          # load image
+          podman load -i crc-e2e-linux-arm64.tar
+          # run
+          cmd="crc-qe/run.sh -junitFilename crc-e2e-junit.xml -targetFolder crc-qe"
+          cmd_microshift="${cmd} -e2eTagExpression '@story_microshift'"
+          cmd_openshift="${cmd} -e2eTagExpression '~@minimal && ~@story_microshift'"
+          podman run --rm -d --privileged --name crc-e2e \
+            -e TARGET_HOST=$(cat host) \
+            -e TARGET_HOST_USERNAME=$(cat username) \
+            -e TARGET_HOST_KEY_PATH=/data/id_rsa \
+            -e BASTION_HOST_USERNAME=$(cat bastion_username) \
+            -e BASTION_HOST=$(cat bastion_host) \
+            -e TARGET_FOLDER=crc-qe \
+            -e TARGET_RESULTS=results \
+            -e OUTPUT_FOLDER=/data \
+            -e DEBUG=true \
+            -e SSH_AUTH_SOCK=$(cat ssh_auth_sock) \
+            -v "$(cat ssh_auth_sock):$(cat ssh_auth_sock)" \
+            -v $PWD/pull-secret:/opt/crc/pull-secret:z \
+            -v $PWD:/data:z \
+            quay.io/crcont/crc-e2e:gh-linux-arm64 \
+                ${cmd_openshift} 
+          podman logs -f crc-e2e
+
+      - name: Test Report
+        working-directory: ${{ env.timestamp }}
+        id: test-report
+        uses: mikepenz/action-junit-report@v4
+        if: always() 
+        with:
+          fail_on_failure: true
+          include_passed: true
+          detailed_summary: true
+          require_tests:  true
+          report_paths: '**/*.xml'
+      
+      - name: Upload e2e results
+        working-directory: ${{ env.timestamp }}
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: linux-e2e-test
+          path: |
+            **/*.xml
+            **/*.results
+            **/*.log
+
```

Comment:
It is not required to decommission the provisioned machine? May not but I guess it would be a good practice to reduce cost on their end?